### PR TITLE
Multi-worker deployment compatibility

### PR DESCRIPTION
**Pull request by turntable-justin**
New features:
- job archiving
- falling back to git pull if subtask is run on different worker than main task (and hence pre-populated temp folder is not available)

----

> [!IMPORTANT]
> Add job archiving and fallback for git pull in multi-worker deployments, updating models, views, tasks, and tests.
> 
>   - **Features**:
>     - Add job archiving to `ScheduledWorkflow` and `DBTOrchestrator` models, with `archive()` and `restore()` methods.
>     - Implement fallback to `git pull` if subtask runs on a different worker than the main task in `orchestration.py`.
>   - **Models**:
>     - Add `archived` field to `ScheduledWorkflow` in `0047_scheduledworkflow_archived_and_more.py`.
>     - Modify `periodic_task` field in `ScheduledWorkflow` to `SET_NULL` on delete.
>   - **Views**:
>     - Update `JobViewSet` in `orchestration_views.py` to handle archived jobs, including `restore` action.
>     - Filter archived jobs in `list()` and `retrieve()` methods.
>   - **Tasks**:
>     - Add `CustomTaskNoStatusConsumer` class in `utils.py` to handle tasks without status updates.
>     - Modify `run_dbt_command` and `save_artifacts_task` in `orchestration.py` to use `repo_override`.
>   - **Tests**:
>     - Add tests for archiving and restoring jobs in `test_orchestration_views.py`.
>     - Add test for `long_running_task` in `test_workflow_utils.py`.
>   - **Misc**:
>     - Add `procps` to `Dockerfile` dependencies.
>     - Add `--worker_name` argument to `start_worker.py` and create `stop_worker.py` command.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=turntable-so%2Fturntable&utm_source=github&utm_medium=referral)<sup> for 81796e270ba08eaa0ed68c546d841ebe7f4aa163. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: backend/app/workflows/utils.py [hunk 1]
```diff
@@ -33,10 +34,14 @@ def run_subtasks(self, *tasks):
         if self.request.called_directly:
             return chain(*tasks)._run_directly()
         elif self.request.is_eager:
-            return chain(*tasks).apply(*tasks, parent_task=self).get()
+            return chain(*tasks).apply(parent_task=self).get()
         return chain(*tasks).apply_async(parent_task=self).get()
 
 
+class CustomTaskNoStatusConsumer(CustomTask):
```
Comment (ianrtracey) [inline backend/app/workflows/utils.py:41]: what is this doing?

File: backend/app/workflows/orchestration.py [hunk 2]
```diff
@@ -132,10 +135,14 @@ def save_artifacts_task(
     from app.models.resources import ArtifactSource
     from app.models.workflows import ArtifactType, TaskArtifact
 
+    repo_override = (
+        GitRepo(repo_override_dir) if os.path.exists(repo_override_dir) else None
```
Comment (ianrtracey) [inline backend/app/workflows/orchestration.py:139]: is there a reason we wouldn't want to throw if the path doesnt exist? this has the risk of creating a hard to debug bug

File: backend/app/views/orchestration_views.py [hunk 3]
```diff
@@ -81,6 +99,8 @@ def runs(self, request, pk=None):
     @action(detail=True, methods=["get"])
     def analytics(self, request, pk=None):
         job = DBTOrchestrator.objects.get(id=pk)
+        if job.archived:
```
Comment (ianrtracey) [inline backend/app/views/orchestration_views.py:102]: do we need this? We'll have to change the frontend to handle it properly. is there a reason we'd want to hide the analytics of an archived job?

File: backend/app/views/orchestration_views.py [hunk 4]
```diff
@@ -56,16 +58,32 @@ def environments(self, request):
...
-        DBTOrchestrator.objects.get(id=pk).delete()
+        job = DBTOrchestrator.objects.get(id=pk)
+        if job.archived:
+            return Response(status=status.HTTP_410_GONE)
+        job.archive()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @action(detail=True, methods=["post"])
+    def restore(self, request, pk=None):
+        job = DBTOrchestrator.objects.get(id=pk)
```
Comment (ianrtracey) [inline backend/app/views/orchestration_views.py:75]: do we want people to be able to restore? From my experience, restoring usually has more edge cases than it's worth vs creating from scratch

**Comment by ianrtracey** [comment]
Couple of thoughts:
1. We should develop the frontend integration on this same branch. That would make it easier to test the feature / isolate the code changes and keep the deploys atomic.
2. We landed on archiving but are we sure this is the right approach? What is the use case for archiving/restoring a job? The downside of this approach is it makes our code more complicated (many checks for is_archived) that we'll have to keep in mind for all future orchestration features as well.

File: backend/app/workflows/utils.py (hunk 1 above)
Comment (turntable-justin) [reply backend/app/workflows/utils.py:41]: Right now, we send workflow status signals based on the workspace_id. For situations where you don't want this, this is used as a signal

File: backend/app/workflows/orchestration.py (hunk 2 above)
Comment (turntable-justin) [reply backend/app/workflows/orchestration.py:139]: This is the error we talked about on the call. The issue this is fixing would occur if we have two worker machines and the parent and the child are allocated to different machines.

File: backend/app/views/orchestration_views.py (hunk 4 above)
Comment (turntable-justin) [reply backend/app/views/orchestration_views.py:75]: I'm open to abandoning this approach. I agree it would be simpler. The main downside is that if someone deletes a workflow, the results would be gone forever.

**Review by ianrtracey — APPROVED** [review]
